ingress-nginx.jsonnet

```diff
@@ -0,0 +1,202 @@
+        verbs: ["get"],
+      },
+      {
+        apiGroups: [""],
+        resources: ["services"],
+        verbs: ["get", "list", "watch"],
+      },
+      {
+        apiGroups: ["extensions"],
+        resources: ["ingresses"],
+        verbs: ["get", "list", "watch"],
+      },
+      {
+        apiGroups: ["extensions"],
+        resources: ["ingresses/status"],
+        verbs: ["update"],
+      },
+      {
+        apiGroups: [""],
+        resources: ["events"],
+        verbs: ["create", "patch"],
+      },
+    ],
+  },
+
+  ingressControllerRole: kube.Role("nginx-ingress-controller") + $.namespace {
+    rules: [
+      {
+        apiGroups: [""],
+        resources: ["configmaps", "pods", "secrets", "namespaces"],
+        verbs: ["get"],
+      },
+      {
+        apiGroups: [""],
+        resources: ["configmaps"],
+        local election_id = "ingress-controller-leader",
+        local ingress_class = "nginx",
+        resourceNames: ["%s-%s" % [election_id, ingress_class]],
+        verbs: ["get", "update"],
+      },
+      {
+        apiGroups: [""],
+        resources: ["configmaps"],
+        verbs: ["create"],
+      },
+      {
+        apiGroups: [""],
+        resources: ["endpoints"],
+        verbs: ["get"], // ["create", "update"],
+      },
+    ],
+  },
+
+  ingressControllerClusterRoleBinding: kube.ClusterRoleBinding("nginx-ingress-controller") {
+    roleRef_: $.ingressControllerClusterRole,
+    subjects_: [$.serviceAccount],
+  },
+
+  ingressControllerRoleBinding: kube.RoleBinding("nginx-ingress-controller") + $.namespace {
+    roleRef_: $.ingressControllerRole,
+    subjects_: [$.serviceAccount],
+  },
+
+  serviceAccount: kube.ServiceAccount("nginx-ingress-controller") + $.namespace,
+
+  service: kube.Service("nginx-ingress") + $.namespace {
+    local this = self,
+    target_pod: $.controller.spec.template,
+    spec+: {
+      ports: [
+        {name: "http", port: 80, protocol: "TCP"},
+        {name: "https", port: 443, protocol: "TCP"},
+      ],
+      type: "LoadBalancer",
+    },
+  },
+
+  controller: kube.Deployment("nginx-ingress-controller") + $.namespace {
+    spec+: {
+      template+: {
+        metadata+: {
+          annotations+: {
+            "prometheus.io/scrape": "true",
+            "prometheus.io/port": "10254",
+            "prometheus.io/path": "/metrics",
+          },
+        },
+        spec+: {
+          serviceAccountName: $.serviceAccount.metadata.name,
+          //hostNetwork: true, // access real source IPs, IPv6, etc
+          terminationGracePeriodSeconds: 60,
+          containers_+: {
+            default: kube.Container("nginx") {
+              image: "gcr.io/google_containers/nginx-ingress-controller:0.9.0-beta.15",
+              env_+: {
+                POD_NAME: kube.FieldRef("metadata.name"),
+                POD_NAMESPACE: kube.FieldRef("metadata.namespace"),
+              },
+              command: ["/nginx-ingress-controller"],
```

Comment:
we talked about running the controller with a custom `--ingress-class` to distinguish it from any other nginx ingress installed in the cluster. WDYT?